source/developer-guides/boot-flow.rst

```diff
@@ -36,6 +36,118 @@ Payload    Load, verify and launch OS images; or perform firmware update
   If verification fails, one of two things occurs. If resiliency is enabled, the boot is swapped over to the alternate partition
   to recover the broken partition (see :ref:`firmware-resiliency-and-recovery`). Otherwise, the boot is halted.
 
+Stage 1A
+~~~~~~~~
+
+When coming out of reset, the processor may be in a very raw state and needs some basic
+initialization before the rest of the platform can be initialized. Stage 1A sets up the
+initial exectution environment to support the execution of the following stages. At this
+stage, a stack is not available and hence assembly used to carry out the basic initialization

```

Comment:
After coming out of reset, when the processor starts executing from the reset vector, there may not be writeable memory and a stack might have not been setup. So, the reset vector code is typically implemented using assembly language instructions. The lack of writeable memory also prevents usage of global variables.